text/0003-font-types.md

```diff
@@ -0,0 +1,408 @@
+- 2022-01-17
+- status: proposal
+- PR: https://github.com/googlefonts/oxidize/pull/3
+
+## A foundation for building font tooling in Rust
+
+
+# Overview
+
+We propose a Rust library (crate) for zero-allocation access and direct manipulation
+of font tables and types, leveraging Rust’s procedural macro system. This
+library is intended to be an opinion-free transparent representation of the types
+in the specification, to serve as a foundational component for anything that
+needs to read or write font files. An additional goal is to provide a common set
+of types to represent common data formats (tags, glyph identifiers, etc) that
+can be used throughout the broader ecosystem.
+
+
+# Motivation
+
+This is intended to answer some of the questions outlined in this repository's
+README. In particular, we aim to determine whether the following goals are feasible:
+
+- using macros to describe the shape of tables
+-copy generating both zero-allocation read-only table definitions as well as owned,
+  mutable definitions (for compilation)
+- suitable for prototyping & extensibility, i.e. it should be easy to add a new
+  table type for a project without needing to patch the core crate or maintain a
+  fork.
+
+# Design
+
+The particulars of the design will be determined through this exploration, but a
+few core ideas are clear. For the read-only case, the design is largely borrowed
+directly from [pinot][], with the main difference being using macros for code
+generation, and that approach has been successful for [fonttools-rs][].
+
+## conversion to and from big-endian bytes
+
+Values in the binary tables are generally big-endian and may not be aligned to
```

Comment:
Are they sometimes *not* big-endian?

Reply:
Suggest citing https://docs.microsoft.com/en-us/typography/opentype/spec/otff#data-types wrt big-endian'ness

Reply:
> Are they sometimes _not_ big-endian?

when the values are u8s maybe it becomes sort of an epistemological question, but agreed that 'generally' is unhelpful.